### PR TITLE
Shahan/onboarding/survey

### PR DESCRIPTION
**Pull request by ShahanM**
Added the final onboarding step to collect initial demographic information.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -264,5 +266,58 @@ def get_pref(topic):
     )
 
 
+@app.route(f"{URL_PREFIX}/onboarding", methods=["GET", "POST"])
+@auth.requires_login
+def onboarding_survey():
```
Comment (karlhigley) [inline app.py:271]: I might call this something like `demographic_survey` to leave room for future expansion of the things we ask on sign-up

**Comment by karlhigley** [comment]
I'll defer to @kluver for reviewing the web code, but generally looks good I think

File: app.py (hunk 1 above)
Comment (ShahanM) [reply app.py:266]: Alright, I changed that URL to `demographic_survey`.

File: app.py [hunk 2]
```diff
@@ -264,5 +266,58 @@ def get_pref(topic):
...
+        return redirect(url_for("home"))
+
+    if request.method == "POST":
+        print(request.form)
+
+        with DB_ENGINE.connect() as conn:
+            repo = DbDemographicsRepository(conn)
+            account_id = auth.get_account_id()
+            
+            demo = Demographics(
```
Comment (kluver) [inline app.py:277]: We should probably do some validation here. I don't expect issues, but realistically if people are sending weird data with postman or html-editing we should handle it without crashing.

File: app.py [hunk 3]
```diff
@@ -264,5 +266,58 @@ def get_pref(topic):
...
+                                group_id=None,
+                                endpoint_url=DEFAULT_RECS_ENDPOINT_URL,
+                            )
+                return redirect(url_for("home", error_description="You have been subscribed!"))
+
+
+
+    today = datetime.date.today()
+    oneyear = timedelta(days=365)
+    yearmin = (today - 100 * oneyear).year # arbitrarilly set to 100 years old
```
Comment (kluver) [inline app.py]: Let's bump this to 123 years: https://en.wikipedia.org/wiki/List_of_the_verified_oldest_people#:~:text=The%20oldest%20person%20ever%20whose,122%20years%20and%20164%20days.

File: templates/onboarding_survey.html [hunk 4]
```diff
@@ -0,0 +1,123 @@
+{% extends "layout.html" %}
+{% block header %}
+<script>
+    document.addEventListener('DOMContentLoaded', function () {
+
+        // FIXME: hardercoded index
+        var racecheckboxes = document.querySelectorAll('input[type="checkbox"]');
```
Comment (kluver) [inline templates/onboarding_survey.html:7]: This should work.

Wouldn't it have been easier to have a `type=hidden` field that you just flip to not hidden as needed?

File: templates/onboarding_survey.html [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+    });
+
+    function validateForm() {
+        var checkboxes = document.querySelectorAll('input[type="checkbox"]');
+
+        var allcheckboxes = Array();
+        for (var i = 0; i < checkboxes.length; i++) {
+            console.log(checkboxes[i]);
+            if (checkboxes[i].checked) {
```
Comment (kluver) [inline templates/onboarding_survey.html:39]: couldn't you return true here? then if you exit the loop unconditionally return false?

File: templates/onboarding_survey.html [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            }
+        }
+        if (allcheckboxes.length < 1) {
+            return false;
+        }
+    }
+</script>
+{% endblock %}
+{% block content %}
+<div class="column box onboarding-survey">
```
Comment (kluver) [inline templates/onboarding_survey.html:50]: note -- form not carefully reviewed. It all looks groovy on a quick pass and I trust your html

**Review by kluver — APPROVED** [review]
Some things I would like to see changed, nothing that's a show-stopper, mostly just comments, not concerns.

**Comment by kluver** [comment]
Hold on, actually, one thing. I didn't see any changes to `auth.py`  I would have expected `STATUS_REDIRECTS` (line 22) to get more content for the new onboarding-style user-status you added.

File: templates/onboarding_survey.html (hunk 4 above)
Comment (ShahanM) [reply templates/onboarding_survey.html:7]: I agree with that. I will make that change. I completely forgot about `type=hidden`

File: templates/onboarding_survey.html (hunk 5 above)
Comment (ShahanM) [reply templates/onboarding_survey.html:29]: Yes I can. Wow, that's embarrassing.

Done.

File: app.py (hunk 3 above)
Comment (ShahanM) [reply app.py]: Done.

File: app.py (hunk 2 above)
Comment (ShahanM) [reply app.py]: Done

**Comment by ShahanM** [comment]
Made all the suggested changes and verified that the entire onboarding flow works.

**Pull request merged**